pkg/automationconfig/automation_config_builder.go

```diff
@@ -0,0 +1,89 @@
+package automationconfig
+
+import (
+	"fmt"
+	"github.com/spf13/cast"
+)
+
+type Topology string
+
+const (
+	ReplicaSetTopology Topology = "ReplicaSet"
+)
+
+type Builder struct {
+	processes      []Process
+	replicaSets    []ReplicaSet
+	version        string
+	auth           Auth
+	members        int
+	domain         string
+	name           string
+	topology       Topology
+	mongodbVersion string
+}
+
+func NewBuilder() *Builder {
+	return &Builder{
+		processes:   []Process{},
+		replicaSets: []ReplicaSet{},
+	}
+}
+
+func (b *Builder) SetTopology(topology Topology) *Builder {
+	b.topology = topology
+	return b
+}
+
+func (b *Builder) SetMembers(members int) *Builder {
+	b.members = members
+	return b
+}
+
+func (b *Builder) SetDomain(domain string) *Builder {
+	b.domain = domain
+	return b
+}
+
+func (b *Builder) SetName(name string) *Builder {
+	b.name = name
+	return b
+}
+
+func (b *Builder) SetMongoDBVersion(version string) *Builder {
+	b.mongodbVersion = version
+	return b
+}
+
+func (b *Builder) SetAutomationConfigVersion(version string) *Builder {
```

Comment:
version should be an int, even if it will be transformed into a string by the automationconfig, that's a detail that the caller don't care about. a numeric attribute should be represented by a number

Reply:
very good point, will update 👍